server/registry/internal/storage/save.go

```diff
@@ -80,3 +92,23 @@ func (c *Client) save(ctx context.Context, v interface{}) error {
 	}).Create(v).Error
 	return grpcErrorForDBError(ctx, err)
 }
+
+func (c *Client) saveWithMask(ctx context.Context, v interface{}, fieldMask []string) error {
+	op := c.db.WithContext(ctx).
+		Select(fieldMask).
+		Clauses(clause.Returning{})
+	err := op.Save(v).Error
+	if err == nil && op.RowsAffected == 0 {
+		err = status.Errorf(codes.NotFound, "%s not found in database", v)
+	}
+	return grpcErrorForDBError(ctx, err)
+
+	// op := c.db.WithContext(ctx).
+	// 	Select(fieldMask).
+	// 	Clauses(
+	// 		clause.Returning{},
+	// 		clause.OnConflict{UpdateAll: true},
+	// 	)
+	// err := op.Create(v).Error
+	// return grpcErrorForDBError(ctx, err)
```

Comment:
Did you intend on checking this in? IMO we shouldn't check in code that's commented out.